int16 idx
float64 x
float64 y
bool occluded
